include/deal.II/differentiation/ad/ad_helpers.h

```diff
@@ -0,0 +1,683 @@
+                              const bool ensure_persistent_setting = true);
+
+      //@}
+
+      /**
+       * @name Operations specific to taped mode: Recording tapes
+       */
+      //@{
+
+      /**
+       * Reset the state of the helper class.
+       *
+       * When an instance of an ADHelperBase is stored as a class member object
+       * with the intention to reuse its instance, it may be necessary to reset
+       * the state of the object before use. This is because, internally, there
+       * is error checking performed to ensure that the correct
+       * auto-differentiable data is being tracked and used only when
+       * appropriate. This function clears all member data and, therefore,
+       * allows the state of all internal flags to be safely reset to their
+       * initial state.
+       *
+       * In the rare case that the number of independent or dependent variables
+       * has changed, this can also reconfigured by passing in the appropriate
+       * arguments to the function.
+       *
+       * @note This also resets the active tape number to an invalid number, and
+       * deactivates the recording mode for taped variables.
+       */
+      virtual void
+      reset(const unsigned int n_independent_variables = 0,
+            const unsigned int n_dependent_variables   = 0,
+            const bool         clear_registered_tapes  = true);
+
+      /**
+       * Returns whether or not this class is tracking calculations performed
+       * with its marked independent variables.
+       */
+      bool
+      is_recording() const;
+
+      /**
+       * Returns the tape number which is currently activated for recording or
+       * reading.
+       */
+      types::tape_index
+      active_tape() const;
+
+      /**
+       * Returns whether or not a tape number has already been used
+       * or registered.
+       */
+      bool
+      is_registered_tape(const types::tape_index &tape_index) const;
+
+      /**
+       * Set the buffer sizes for the next active tape.
+       *
+       * This function must be called before enable_record_sensitivities() for
+       * it to have any influence on the memory allocated to the next recorded
+       * tape.
+       *
+       * @note This only has an effect when using ADOL-C numbers. As stated by
+       * the ADOL-C manual, it may be desirable to create a file ".adolcrc" in
+       * the program run directory and set the buffer size therein.
+       * Alternatively, this function can be used to override the settings for
+       * any given tape, or can be used in the event that no ".adolcrc" file
+       * exists.
+       *
+       * @param[in] obufsize ADOL-C operations buffer size
+       * @param[in] lbufsize ADOL-C locations buffer size
+       * @param[in] vbufsize ADOL-C value buffer size
+       * @param[in] tbufsize ADOL-C Taylor buffer size
+       */
+      void
+      set_tape_buffer_sizes(const types::tape_buffer_sizes obufsize = 67108864,
```

Comment:
where do these defaults come from? might be useful to explain the default values in the documentation above